File: src/src/main/java/hackerrank/helper/InputStream.java
package hackerrank.helper;

import java.util.LinkedList;
import java.util.List;

public class InputStream {

    private LinkedList<String> _input = new LinkedList<>();

    private String getNext() {
        if (_input.isEmpty()) return "";
        else return _input.pop();
    }

    public void setInputData(List<String> in) {
        in.forEach((x) -> _input.add(x));
    }

    public String getNextString() {
        return this.getNext();
    }

    public Integer getNextInt() {
        return Integer.parseInt(this.getNext());
    }
}

File: src/src/main/scala/hackerrank/helper/InputStream.scala
package hackerrank.helper

import scala.collection.mutable

class InputStream {

  private val input: mutable.Queue[String] = mutable.Queue()
  private def getNext: String = if (input.isEmpty) "" else input.dequeue

  def setInputData(in: Seq[String]) = in.foreach(x => input.enqueue(x))
  def getNextString(): String = this.getNext
  def getNextInt(): Int = {
    val result = input.front.split(" ", 2)
    if (result.tail.nonEmpty) input.update(0, result.tail.head)
    else input.dequeue()
    result.head.toInt
  }
  def close() = input.clear()
}
